webapp/src/create-transaction/index.js

```diff
@@ -0,0 +1,96 @@
+import React, { useState } from 'react'
+import gql from 'graphql-tag'
+import { useMutation } from '@apollo/react-hooks'
+import { withRouter } from 'react-router'
+import { css } from '@emotion/core'
+
+const CREATE_TRANSACTION = gql`
+  mutation AddTransaction(
+    $userId: String
+    $description: String
+    $merchantId: String
+    $isDebit: Boolean
+    $isCredit: Boolean
+    $amount: Float
+  ) {
+    addTransaction(
+      user_id: $userId
+      description: $description
+      merchant_id: $merchantId
+      debit: $isDebit
+      credit: $isCredit
+      amount: $amount
+    ) {
+      user_id
+      description
+      merchant_id
+      debit
+      credit
+      amount
+      id
+    }
+  }
+`
+
+export const CreateTransaction = () => {
+  const [createTransaction] = useMutation(CREATE_TRANSACTION)
+  const [amount, setAmount] = useState('')
+  const [isCredit, setIsCredit] = useState(false)
+  const [isDebit, setIsDebit] = useState(false)
+  const [userId, setUserId] = useState('')
+  const [merchantId, setMerchantId] = useState('')
+  const [description, setDescription] = useState('')
+
+  return (
+    <form css={formStyle} onSubmit={(event) => {
```

Comment:
Should use formik for validation. Don't allow form submission until fields are filled in.

Reply:
Question for PM: Is only credit or debit allowed for a transaction type?

If so I need to either validate that only one is selected, or maybe use a select instead of checkboxes